apps/bfd-pipeline/bfd-pipeline-rif-load/src/main/java/gov/cms/bfd/pipeline/rif/load/RifLoaderIdleTasks.java

```diff
@@ -0,0 +1,355 @@
+
+  /** JPQL query for a count */
+  private static final String COUNT_UNHASHED_BENFICIARIES =
+      "select count(b) from Beneficiary b "
+          + "where b.mbiHash is null and b.medicareBeneficiaryId is not null and "
+          + "b.medicareBeneficiaryId is not empty";
+
+  /** JPQL query for a list */
+  private static final String SELECT_UNHASHED_HISTORIES =
+      "select b from BeneficiaryHistory b "
+          + "where b.mbiHash is null and b.medicareBeneficiaryId is not null and "
+          + "b.medicareBeneficiaryId is not empty";
+
+  /** JPQL query for a count */
+  private static final String COUNT_UNHASHED_HISTORIES =
+      "select count(b) from BeneficiaryHistory b "
+          + "where b.mbiHash is null and b.medicareBeneficiaryId is not null and "
+          + "b.medicareBeneficiaryId is not empty";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(RifLoaderIdleTasks.class);
+
+  /** Enum to tell what the current task is being executed. */
+  public enum Task {
+    /** The initial task which is executed after startup */
+    INITIAL,
+
+    /** After the initial task, this task is run. */
+    POST_STARTUP,
+
+    /** Run the normal task */
+    NORMAL,
+  }
+
+  /* Hashing entities */
+  private final LoadAppOptions options;
+  private final EntityManagerFactory entityManagerFactory;
+  private final SecretKeyFactory secretKeyFactory;
+
+  /* Metrics */
+  private final Meter beneficaryMeter;
+  private final Meter historyMeter;
+
+  /* Thread pool for post startup tasks */
+  private final ExecutorService executorService;
+
+  /* The task that is going to execute next */
+  private Task currentTask = Task.INITIAL;
+
+  /**
+   * Create a helper to manage the idle time tasks.
+   *
+   * @param options pipeline options
+   * @param appMetrics pipeline metrics
+   * @param entityManagerFactory a connection to the database of the pipeline
+   * @param secretKeyFactory for hashing
+   */
+  public RifLoaderIdleTasks(
+      final LoadAppOptions options,
+      final MetricRegistry appMetrics,
+      final EntityManagerFactory entityManagerFactory,
+      final SecretKeyFactory secretKeyFactory) {
+    this.options = options;
+    this.entityManagerFactory = entityManagerFactory;
+    this.secretKeyFactory = secretKeyFactory;
+
+    this.beneficaryMeter = appMetrics.meter("fixups.beneficiary.rate");
+    this.historyMeter = appMetrics.meter("fixups.beneficiary_history.rate");
+
+    this.executorService = Executors.newFixedThreadPool(THREAD_COUNT);
+  }
+
+  /**
+   * What task will be executed in the next idle time slot
+   *
+   * @return the current task
+   */
+  public Task getCurrentTask() {
+    return currentTask;
+  }
+
+  /**
+   * Run the current idle task. This method is expected to be called whenever no RIF files are
+   * present for process. It will respect the TASK_TIME_LIMIT to allow checking of RIF files and to
+   * not interfer with RIF file processing.
+   */
+  public void doIdleTask() {
```

Comment:
Suggest renaming to `doCurrentTask()` or something similar that better describes the action being taken.